web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js

```diff
@@ -288,11 +288,24 @@ odoo.define('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer', function (requ
             }
             // Enforce mode of the parent
             options.mode = this.getParent().mode;
-            var widget = this._renderFieldWidget(
-                node, record, _.pick(options, 'mode')
-            );
-            this._handleAttributes(widget.$el, node);
-            return $td.append(widget.$el);
+            if (node.tag === 'widget') {
+                return $td.append(this._renderWidget(record, node));
+            }
+            if (node.attrs.widget || (options && options.renderWidgets)) {
+                var $el = this._renderFieldWidget(node, record, _.pick(options, 'mode'));
+                this._handleAttributes($el, node);
+                return $td.append($el);
+            }
+            var name = node.attrs.name;
+            var field = this.state.fields[name];
+            var value = record.data[name];
+            var formattedValue = field_utils.format[field.type](value, field, {
```

Comment:
field.type must be node.attrs.widget, right? So float_time format etc can be used.

Reply:
I would be glad to answer, yet I didn't even get what you've asked about - yet code seems ok, taking into account that list_renderer was always a base for 2d_matrix_renderer

Reply:
https://github.com/odoo/odoo/blob/2a4c44bb4f0ef0d02c62991f588c20c0741616e2/addons/web/static/src/js/views/list/list_renderer.js#L301

Reply:
@tarteo ^

Reply:
@tarteo nope, the format utils works at field type level (m2o, char, etc) so: this sounds correct :+1:

Reply:
@simahawk There's also float_time in format. https://github.com/odoo/odoo/blob/d40956c7909bffa338fd0138565044571e852da3/addons/web/static/src/js/fields/field_utils.js#L660